pymc/distributions/timeseries.py

```diff
@@ -237,64 +297,39 @@ def logp(self, x):
         -------
         TensorVariable
         """
-        if x.ndim > 0:
-            x_im1 = x[:-1]
-            x_i = x[1:]
-            mu, sigma = self._mu_and_sigma(self.mu, self.sigma)
-            innov_like = Normal.dist(mu=x_im1 + mu, sigma=sigma).logp(x_i)
-            return self.init.logp(x[0]) + at.sum(innov_like)
-        return self.init.logp(x)
+        import pymc as pm
 
-    def random(self, point=None, size=None):
-        """Draw random values from GaussianRandomWalk.
+        # Implement using AePPL
+        # I need to create a graph that calculates the logp of GRW
+        # I can use AePPL or PyMC to do it
 
-        Parameters
-        ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
+        def normal_logp(value, mu, sigma):
+            logp = (
+                -0.5 * at.pow((value - mu) / sigma, 2)
+                - at.log(at.sqrt(2.0 * np.pi))
+                - at.log(sigma)
+            )
+            return logp
 
-        Returns
-        -------
-        array
-        """
-        # sigma, mu = distribution.draw_values([self.sigma, self.mu], point=point, size=size)
-        # return distribution.generate_samples(
-        #     self._random,
-        #     sigma=sigma,
-        #     mu=mu,
-        #     size=size,
-        #     dist_shape=self.shape,
-        #     not_broadcast_kwargs={"sample_shape": to_tuple(size)},
-        # )
-        pass
-
-    def _random(self, sigma, mu, size, sample_shape):
-        """Implement a Gaussian random walk as a cumulative sum of normals.
-        axis = len(size) - 1 denotes the axis along which cumulative sum would be calculated.
-        This might need to be corrected in future when issue #4010 is fixed.
+        # Create logp calculation graph the inital time point
+        innit_logp = normal_logp(value[0] - init, mu, sigma)
+
+        # Create logp calculation graph for innovations
+        stationary_vals = at.diff(value[1:]) - init
```

Comment:
I don't think init matters anymore here. Also I think the diff should include the whole array. I could be mistaken

Reply:
youre right, init is not needed. 

Diff returns an array thats len(n) -1, fixed the `[1:]` indexing as such

![image](https://user-images.githubusercontent.com/7213793/149548784-9ea32bed-a964-4f84-a801-9fc9ef34343f.png)